tests/build-script.rs

```diff
@@ -1565,6 +1565,82 @@ fn cfg_override_doc() {
 }
 
 #[test]
+fn env_build() {
+    let build = project("builder")
+        .file("Cargo.toml", r#"
+            [package]
+            name = "builder"
+            version = "0.0.1"
+            authors = []
+            build = "build.rs"
+        "#)
+        .file("src/main.rs", r#"
+            const FOO: &'static str = env!("FOO");
+            fn main() {}
+        "#)
+        .file("build.rs", r#"
+            fn main() {
+                println!("cargo:rustc-env=FOO=foo");
+            }
+        "#);
+    assert_that(build.cargo_process("build").arg("-v"),
+                execs().with_status(0));
+}
+
+#[test]
+fn env_test() {
```

Comment:
The problem is that this test still fails at the build step (with `FOO` variable not being defined). There is apparently something more that needs to be done to carry the variables if the build scripts is invoked for a crate as dependency for the test crate.